src/rnp/fficli.cpp

```diff
@@ -419,11 +419,13 @@ ffi_pass_callback_stdin(rnp_ffi_t        ffi,
             ok = true;
         }
 
-        rnp_buffer_clear(rnp->reused_password, strnlen(rnp->reused_password, buf_len));
-        free(rnp->reused_password);
-        rnp->reused_password = NULL;
-        rnp->reuse_password_for_subkey = false;
-        rnp_buffer_destroy(primary_fprint);
+        rnp->reuse_password_for_subkey--;
+        if (rnp->reuse_password_for_subkey == 0) {
```

Comment:
We are stick to `(!rnp->reuse_password_for_subkey)` in such cases (however there could be some artifacts left from older codebase).